concordium-consensus/src/Concordium/Scheduler.hs

```diff
@@ -1196,15 +1197,69 @@ handleContractUpdateV1 originAddr istance checkAndGetSender transferAmount recei
                                         -- Charge for updating the pointer in the instance to the new module.
                                         tickEnergy Cost.initializeContractInstanceCreateCost
                                         go (resumeEvent True:upgradeEvent:interruptEvent:events) =<< runInterpreter (return . WasmV1.resumeReceiveFun rrdInterruptedConfig rrdCurrentState False newBalance WasmV1.Success Nothing)
-
+                  WasmV1.QueryAccountBalance {..}  -> do
+                    newBalance <- getCurrentContractAmount Wasm.SV1 istance
+                    -- TODO Charge query cost or maybe not
+                    -- Lookup account
+                    maybeAccount <- getStateAccount imqabAddress
+                    case maybeAccount of
+                      Nothing ->
+                        go events =<< runInterpreter (return . WasmV1.resumeReceiveFun rrdInterruptedConfig rrdCurrentState False
+                             newBalance (WasmV1.Error $ WasmV1.EnvFailure $ WasmV1.MissingAccount imqabAddress) Nothing)
+                      Just (_accountIndex, account) -> do
+                        -- Lookup account balances
+                        balance <- getAccountAmount account
+                        accountStake <- getAccountStake account
+                        let stake = case accountStake of
+                              AccountStakeNone -> 0
+                              AccountStakeBaker bkr -> bkr ^. stakedAmount
+                              AccountStakeDelegate del -> del ^. delegationStakedAmount
+                        lockedAmount <- do schedule <- getAccountReleaseSchedule account
+                                           return $ schedule ^. totalLockedUpBalance
```

Comment:
I did consider copying from the PR, but part of it uses the restructuring and this will probably produce less conflicts.
I could add a TODO for replacing it?

Reply:
These kinds of todos never get resolved :)